src/AppInstallerCLICore/PackageCollection.cpp

```diff
@@ -0,0 +1,187 @@
+            root[s_PackagesJson_Schema] = s_PackagesJson_SchemaUri_V0;
+
+            // TODO: This uses localtime. Do we want to use UTC or add time zone?
+            std::stringstream currentTimeStream;
+            Utility::OutputTimePoint(currentTimeStream, std::chrono::system_clock::now());
+            root[s_PackagesJson_CreationDate] = currentTimeStream.str();
+
+            return root;
+        }
+
+        // Adds a new Package node to a Source node in the Json file, and returns it.
+        Json::Value& AddPackageToSource(Json::Value& sourceNode, const PackageCollection::Package& package)
+        {
+            Json::Value packageNode{ Json::ValueType::objectValue };
+            packageNode[s_PackagesJson_Package_Id] = package.Id.get();
+            packageNode[s_PackagesJson_Package_Version] = package.VersionAndChannel.GetVersion().ToString();
+
+            // Only add channel if present
+            const std::string& channel = package.VersionAndChannel.GetChannel().ToString();
+            if (!channel.empty())
+            {
+                packageNode[s_PackagesJson_Package_Channel] = channel;
+            }
+
+            return sourceNode[s_PackagesJson_Packages].append(std::move(packageNode));
+        }
+
+        // Adds a new Source node to the JSON, and returns it.
+        Json::Value& AddSourceNode(Json::Value& root, const PackageCollection::Source& source)
+        {
+            Json::Value sourceNode{ Json::ValueType::objectValue };
+
+
+            Json::Value sourceDetailsNode{ Json::ValueType::objectValue };
+            sourceDetailsNode[s_PackagesJson_Source_Name] = source.Details.Name;
+            sourceDetailsNode[s_PackagesJson_Source_Argument] = source.Details.Arg;
+            sourceDetailsNode[s_PackagesJson_Source_Identifier] = source.Details.Identifier;
+            sourceDetailsNode[s_PackagesJson_Source_Type] = source.Details.Type;
+            sourceNode[s_PackagesJson_Source_Details] = std::move(sourceDetailsNode);
+
+            sourceNode[s_PackagesJson_Packages] = Json::Value{ Json::ValueType::arrayValue };
+
+            auto& sourcesNode = GetJsonProperty(root, s_PackagesJson_Sources, Json::ValueType::arrayValue);
+            for (const auto& package : source.Packages)
+            {
+                AddPackageToSource(sourceNode, package);
+            }
+
+            return sourcesNode.append(std::move(sourceNode));
+        }
+    }
+
+    namespace PackagesJson
+    {
+        Json::Value CreateJson(const PackageCollection& packages)
+        {
+            Json::Value root = CreateRoot(packages.ClientVersion);
+            for (const auto& source : packages.Sources)
+            {
+                AddSourceNode(root, source);
+            }
+
+            return root;
+        }
+
+        std::optional<PackageCollection> ParseJson(const Json::Value& root)
+        {
+            // TODO: Embed schema in binaries & validate file
```

Comment:
I had some trouble getting this to work. @yao-msft is also working on something with JSON schemas and seems to have this figured out, so I'll wait for him to check in his change and then hook this up with his code.